controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -178,6 +176,18 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) resetCephResources() {
+	r.cephResourcesByName = map[string]*ocsv1alpha1.CephResourcesSpec{}
+	for i := 0; i < len(r.storageConsumer.Status.CephResources); i++ {
+		if r.storageConsumer.Status.CephResources[i].Name == r.cephClientHealthChecker.Name {
```

Comment:
This is a very specific check at a very generalized func and It will fail if we will decided to add other resources to the status again in future releases.

A better way is to track what was and wasn't reconciled during the reconciliation and then recreate the status from that info